### ISSUE TITLE
Hooks in Components error

### ISSUE THREAD

**Issue by bahtou** [reharik/redux-task-calendar#1]
Seems like no hook is able to work in the Calendar container component and Im not certain why

`import { Calendar } from "./../../build/redux-task-calendar";`

something with the bundling is not right. Are you sure the development webpacks are the same? I see you have two versions.

**Comment by bahtou** [comment]
from the `index.html` 

<img width="729" alt="Screen Shot 2019-03-10 at 7 11 32 PM" src="https://user-images.githubusercontent.com/1924298/54093675-631cc380-4368-11e9-814c-95c7deaae0df.png">


from the imported `app.css`
<img width="272" alt="Screen Shot 2019-03-10 at 7 12 26 PM" src="https://user-images.githubusercontent.com/1924298/54093690-80ea2880-4368-11e9-9d1f-9536e52b0324.png">


The css classes don't match up.

**Comment by reharik** [comment]
I see it's prefixing the folder name, which I think I knew but it's clearly doing it wrong or like you said reprocrssing it. I think I can make that behave

**Comment by reharik** [comment]
Ok so I figured it out, I needed 
localIdentName: 'redux__task__calendar__[local]',
rather than
localIdentName: '[folder]__[local]',
and then I also needed to add postcss-mixins.
now it's getting css and I can freakin see why the actual calendar doesn't work!

**Comment by bahtou** [comment]
that makes sense
have you looked at `precss` https://www.npmjs.com/package/precss?

**Comment by bahtou** [comment]
question about naming your actions

going through some examples online (like Todo app) and sometimes the naming becomes redundant.
When naming actions that directly affect state, how do you name your actions?  For example:

```
function addTodo() {
    return {
      type: 'ADD_TODO',
      data: blob
    }
}

function addTodo() {
  return {
    type: 'SET_TODO'
    data: blob
  }
}
``` 

and for deleting/removing
```

function deleteTodo() {
    return {
      type: 'DELETE_TODO',
      data: blob
    }
}

function deleteTodo() {
  return {
    type: 'SET_TODO'
    data: blob
  }
}
```

how do you name these things? 
I use `SET_TODO` in both cases above because fundamentally that is what you're doing to the KV store (also similar to redis (get/set) when you interface directly with it)

**Comment by reharik** [comment]
Are you referring to verb? e.g. add, remove?  If so as you may know I'm a
big fan of DDD and in DDD you want to have as much context as possible.  So
I would name them something like well, hell, there's no way around it, Todo
is straight up crud, so I would just pick one and stick with it.  AddTodo,
UpdateTodo, Delete_Todo etc.  Am I getting the question right?
R

**Comment by bahtou** [comment]
i agree   i like to be explicit in these naming to convey the intention behind it

If the domain has a 'widget' for example, then your reducer actions would be something like

```
// Actions
const LOAD   = 'my-app/widgets/LOAD';
const CREATE = 'my-app/widgets/CREATE';
const UPDATE = 'my-app/widgets/UPDATE';
const REMOVE = 'my-app/widgets/REMOVE';
```
or
```
// Actions
const LOAD_WIDGET   = 'LOAD_WIDGET';
const CREATE_WIDGET = 'CREATE_WIDGET';
const UPDATE_WIDGET = 'UPDATE_WIDGET';
const REMOVE_WIDGET = 'REMOVE_WIDGET';
```
??

**Comment by reharik** [comment]
oh, well, I do something similar to the former.
This is how I do it to get away from the horrific boiler plate
https://github.com/reharik/MethodFitness/blob/7bc92212ec565b85fd90ea9a519a72d09b14cb65/frontend/app/src/sagas/requestSaga.js#L22

**Comment by bahtou** [comment]
shouldn't requests (async events) be in the past tense?

REQUESTED, SUCCEEDED, FAILED, COMPLETED

since these actions are outside your control, the application just requests them & the browser/network respond back to the app in the past tense, after they perform the work?

no right/wrong answer    im trying to understand this naming convention

**Comment by reharik** [comment]
Hmmm well are they events? or actions?  If they are actions (commands) they
should be in the imperative. if they are events then the past tense. so
perhaps, it should be request, succeeded and failed. IDK, I mean I think
it's kind of academic.
R

**Comment by bahtou** [comment]
well  its both  and its confusing lol

im starting to do this for async tasks

```
function fetchExternalResource() {
  return { type: FETCH_REQUESTED };
}
```

my function is named in the imperative case (command, something i want to happen which can be rejected) and then my 'event' is something that has happened (fetch requested).

then i have my saga doing a `take(FETCH_REQUESTED)` and emitting an event depending on what happens with it, `put(FETCH_SUCCEEDED)`, etc etc

dunno  i have to figure something out   i am not consistent when i do this in my apps and it's getting to a point where i confuse my self with the english

**Comment by reharik** [comment]
lol, yea, I think you just gotta choose one and shoot!

**Comment by bahtou** [comment]
wrote something up
https://github.com/bahtou/the-book-pundit/wiki/WIP:-Best-Practices-for-naming-Action

what you think?
